### PR TITLE
Replace current errors with custom errors

### PR DESCRIPTION
**Pull request by ivov**
This PR implements the custom error classes created in #1309.

- [x] Replace API errors with `NodeApiError` instances in compatible nodes
- [x] Replace operational errors with `NodeOperationError` instances
- [x] Determine why `NodeOperationError` is incompatible with certain nodes
- [x] Test 20 of nodes with `NodeApiError` and `NodeOperationError`

### Unmodified by `NodeApiError`
- Compression
- Cron
- Crypto
- Datetime
- Edit Image
- Email Read IMAP
- Error Trigger
- Execute Command
- Execute Workflow
- FTP
- Function
- Function Item
- HTML Extract
- HTTP Request
- If
- Interval
- Merge
- MoveBinaryData
- NoOp
- ReadBinaryFile
- ReadBinaryFiles
- Read PDF
- Rename Keys
- RSS Feed Read
- RunAt
- Set
- SplitInBatches
- SpreadsheetFile
- SSE Trigger
- Start
- Switch
- Webhook
- WriteBinaryFile
- XML

### Unmodified by `NodeOperationError`

- Automizy
- Autopilot
- Coingecko
- Discourse
- Egoi
- Google Analytics
- Google Cloud Natural Language
- Google Contacts
- Google Task
- Gotify
- Lemlist
- Mailerlite
- Mailjet
- OpenThesaurus
- PhilipsHue
- PostHog
- Pushcut
- SendGrid
- Snowflake
- Spontit
- Spotify
- Storyblok
- Todoist
- Vonage
- Wise
- Xero
- Zoho
- Cron
- Crypto
- ErrorTrigger
- ExecuteCommand
- Merge
- NoOp
- ReadBinaryFiles
- ReadPdf
- RenameKeys
- RunAt
- Set
- SplitInBatches
- SseTrigger
- Start

### Incompatible with `NodeOperationError`

- ~~Google Sheet~~
- ~~If~~
- ~~MongoDB node~~
- ~~Phantombuster~~
- ~~Redis~~
- ~~Rundeck~~
- ~~Switch~~

Fixed!

### Nodes where `NodeApiError` was tested

- Hacker News
- Wise
- Jira
- QuickBooks
- CoinGecko
- Discourse
- Reddit
- Trello
- Mattermost
- PayPal

### Nodes where `NodeOperationError` was tested

- DateTime
- Function
- Function Item
- HTTP Request
- XML
- HTML Extract
- Compression
- RSS Feed Read
- Write Binary File
- APItemplate.io

### CODE REVIEW

**Comment by ivov** [comment]
@BHesseldieck
- Custom errors testing completed
- Five missed nodes added
- To discuss later:
  - Special handling for [AWS 403 error](https://github.com/n8n-io/n8n/blob/ba7c35dbf5e0c926471a9197a1e221b1ab5474c5/packages/nodes-base/nodes/Aws/Comprehend/GenericFunctions.ts#L64-L74)? (six instances)
  - Special handling for [StravaTrigger error](https://github.com/n8n-io/n8n/blob/34f69fd65231c13942072893726366f817618e82/packages/nodes-base/nodes/Strava/StravaTrigger.node.ts#L163-L189)?
  - Merge to `improve-node-error-handling`?

File: packages/nodes-base/nodes/Aws/SQS/AwsSqs.node.ts [hunk 1]
```diff
@@ -271,7 +272,7 @@ export class AwsSqs implements INodeType {
 					// loads first 1000 queues from SQS
 					data = await awsApiRequestSOAP.call(this, 'sqs', 'GET', `?Action=ListQueues`);
 				} catch (err) {
-					throw new Error(`AWS Error: ${err}`);
+					throw new NodeOperationError(this.getNode(), `AWS Error: ${err}`);
```
Comment (BHesseldieck) [inline packages/nodes-base/nodes/Aws/SQS/AwsSqs.node.ts:275]: don't know what it returns but gut feeling would say that this is rather an API Error than an Operation? what do you think?

File: packages/nodes-base/nodes/Aws/SQS/AwsSqs.node.ts [hunk 2]
```diff
@@ -375,7 +376,7 @@ export class AwsSqs implements INodeType {
 			try {
 				responseData = await awsApiRequestSOAP.call(this, 'sqs', 'GET', `${queuePath}/?Action=${operation}&` + params.join('&'));
 			} catch (err) {
-				throw new Error(`AWS Error: ${err}`);
+				throw new NodeOperationError(this.getNode(), `AWS Error: ${err}`);
```
Comment (BHesseldieck) [inline packages/nodes-base/nodes/Aws/SQS/AwsSqs.node.ts:379]: same as above

File: packages/nodes-base/nodes/Aws/SQS/AwsSqs.node.ts (hunk 1 above)
Comment (ivov) [reply packages/nodes-base/nodes/Aws/SQS/AwsSqs.node.ts]: Good catch! It was an oversight. Fixed.

File: packages/nodes-base/nodes/Aws/SQS/AwsSqs.node.ts (hunk 2 above)
Comment (ivov) [reply packages/nodes-base/nodes/Aws/SQS/AwsSqs.node.ts]: Good catch! It was an oversight. Fixed.

File: packages/nodes-base/nodes/Aws/AwsLambda.node.ts [hunk 3]
```diff
@@ -186,7 +177,7 @@ export class AwsLambda implements INodeType {
 					errorMessage += `\n\nStack trace:\n${responseData.stackTrace}`;
 				}
 
-				throw new Error(errorMessage);
+				throw new NodeOperationError(this.getNode(), errorMessage);
```
Comment (BHesseldieck) [inline packages/nodes-base/nodes/Aws/AwsLambda.node.ts:180]: needs to be an API error here
```suggestion
				throw new NodeApiError(this.getNode(), responseData);
```

**Review by BHesseldieck — CHANGES_REQUESTED** [review]
reviewed all nodes starting with `A`

File: packages/nodes-base/nodes/ActiveCampaign/GenericFunctions.ts [hunk 4]
```diff
@@ -53,7 +53,7 @@ export async function activeCampaignApiRequest(this: IHookFunctions | IExecuteFu
 		const responseData = await this.helpers.request!(options);
 
 		if (responseData.success === false) {
-			throw new Error(`ActiveCampaign error response: ${responseData.error} (${responseData.error_info})`);
+			throw new NodeOperationError(this.getNode(), `ActiveCampaign error response: ${responseData.error} (${responseData.error_info})`);
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/nodes-base/nodes/ActiveCampaign/GenericFunctions.ts:56]: Result with change applied:

image

**Review by BHesseldieck — APPROVED** [review]
🚀

**Pull request merged**